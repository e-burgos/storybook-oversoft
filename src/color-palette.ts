const primary = {
  red: '#D41F34',
  redGradient: `linear-gradient(#D41F3450,#D41F34)`,
  redLight: '#D41F3450',
  redExtraLight: '#D41F3433',
};

const secondary = {
  yellow: '#FAB550',
  yellowGradient: 'linear-gradient(#F2A73A,#E58616)',
  yellowLight: '#FAB55033',
  green: '#1ECE7C',
  greenGradient: 'linear-gradient(#1ECE7C,#21AB6B)',
  greenLight: '#1ECE7C33',
  blue: '#6889FF',
  blueGradient: 'linear-gradient(#6889FF,#2E40E2)',
  blueLight: '#6889FF33',
  violet: '#8D52ED',
  violetGradient: 'linear-gradient(#A87BF0,#7C39EB)',
  violetLight: '#8D52ED33',
};

const neutrals = {
  black: '#000000',
  greyOne: '#333333',
  greyTwo: '#666666',
  greyThree: '#9D9D9D',
  greyLight: '#FAFAFA',
  white: '#FFFFFF',
};

const background = '#FAFAFA';

const colors = { primary, secondary, neutrals, background };

export { colors, primary, secondary, neutrals, background };
export default colors;
