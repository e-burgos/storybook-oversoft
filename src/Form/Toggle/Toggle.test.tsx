import React from 'react';
import ToggleComponent from './Toggle.component';
import { render, fireEvent, act } from '@testing-library/react';
import { primary, neutrals } from '../../color-palette';

test('Render Toggle with default props', async () => {
  const onClickHandlerMock = jest.fn();
  const view = render(<ToggleComponent isToggled={false} onClick={onClickHandlerMock} />);

  expect(view.container).toMatchSnapshot();
});

test('Render disabled Toggle', async () => {
  const onClickHandlerMock = jest.fn();
  const view = render(<ToggleComponent isToggled={false} onClick={onClickHandlerMock} disabled />);

  expect(view.container).toMatchSnapshot();
});

test('onClick Toggle', async () => {
  const onClickHandlerMock = jest.fn();
  const view = render(
    <ToggleComponent
      fontFamily="Rubik-Bold"
      fontSize="12px"
      label="This is the Toggle"
      isToggled={false}
      togglePrimaryColor={primary.red}
      toggleNeutralColor={neutrals.white}
      toggleGuidePrimaryColor={primary.redLight}
      toggleGuideNeutralColor={neutrals.greyThree}
      disabled={false}
      onClick={onClickHandlerMock}
    />
  );

  expect(view.container).toMatchSnapshot();

  await act(() => {
    fireEvent.click(view.getByTestId('This is the Toggle'));
  });

  expect(onClickHandlerMock.mock.calls.length).toBe(1);

  await act(() => {
    view.rerender(
      <ToggleComponent
        fontFamily="Rubik-Bold"
        fontSize="12px"
        label="This is the Toggle"
        isToggled={true}
        togglePrimaryColor={primary.red}
        toggleNeutralColor={neutrals.white}
        toggleGuidePrimaryColor={primary.redLight}
        toggleGuideNeutralColor={neutrals.greyThree}
        disabled={false}
        onClick={onClickHandlerMock}
      />
    );
  });

  expect(view.container).toMatchSnapshot();

  await act(() => {
    fireEvent.click(view.getByTestId('This is the Toggle'));
  });

  expect(onClickHandlerMock.mock.calls.length).toBe(2);
});
