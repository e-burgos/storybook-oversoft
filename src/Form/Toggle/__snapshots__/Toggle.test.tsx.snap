// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Toggle with default props 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  width: 100%;
  opacity: 1;
}

.c1 {
  font-family: Rubik-Bold;
  font-size: 12px;
  margin-right: 8px;
}

.c2 {
  position: relative;
  display: inline-block;
  height: 14px;
  background: #9D9D9D;
  border-radius: 100px;
  width: 42.272px;
}

.c3 {
  flex-shrink: 0;
  fill: var(--system-colors-scl-white, #fff);
  filter: drop-shadow(0px 1.0458389520645142px 2.0916779041290283px rgba(0, 0, 0, 0.2)) drop-shadow(0px 0.10458389669656754px 0.3137516975402832px rgba(0, 0, 0, 0.1));
  border-radius: 50%;
  position: absolute;
  cursor: pointer;
  top: -6px;
  left: -2px;
  bottom: 0;
  background-color: #ccc;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  content: '';
  height: 26px;
  width: 26px;
  background-color: #FFFFFF;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
      font-family="Rubik-Bold"
      font-size="12px"
    />
    <label
      class="c2"
      data-testid=""
    >
      <span
        class="c3"
      />
    </label>
  </div>
</div>
`;

exports[`Render disabled Toggle 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  width: 100%;
  opacity: 0.5;
}

.c1 {
  font-family: Rubik-Bold;
  font-size: 12px;
  margin-right: 8px;
}

.c2 {
  position: relative;
  display: inline-block;
  height: 14px;
  background: #9D9D9D;
  border-radius: 100px;
  width: 42.272px;
}

.c3 {
  flex-shrink: 0;
  fill: var(--system-colors-scl-white, #fff);
  filter: drop-shadow(0px 1.0458389520645142px 2.0916779041290283px rgba(0, 0, 0, 0.2)) drop-shadow(0px 0.10458389669656754px 0.3137516975402832px rgba(0, 0, 0, 0.1));
  border-radius: 50%;
  position: absolute;
  cursor: pointer;
  top: -6px;
  left: -2px;
  bottom: 0;
  background-color: #ccc;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  content: '';
  height: 26px;
  width: 26px;
  background-color: #FFFFFF;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
      font-family="Rubik-Bold"
      font-size="12px"
    />
    <label
      class="c2"
      data-testid=""
    >
      <span
        class="c3"
      />
    </label>
  </div>
</div>
`;

exports[`onClick Toggle 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  width: 100%;
  opacity: 1;
}

.c1 {
  font-family: Rubik-Bold;
  font-size: 12px;
  margin-right: 8px;
}

.c2 {
  position: relative;
  display: inline-block;
  height: 14px;
  background: #9D9D9D;
  border-radius: 100px;
  width: 42.272px;
}

.c3 {
  flex-shrink: 0;
  fill: var(--system-colors-scl-white, #fff);
  filter: drop-shadow(0px 1.0458389520645142px 2.0916779041290283px rgba(0, 0, 0, 0.2)) drop-shadow(0px 0.10458389669656754px 0.3137516975402832px rgba(0, 0, 0, 0.1));
  border-radius: 50%;
  position: absolute;
  cursor: pointer;
  top: -6px;
  left: -2px;
  bottom: 0;
  background-color: #ccc;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  content: '';
  height: 26px;
  width: 26px;
  background-color: #FFFFFF;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
      font-family="Rubik-Bold"
      font-size="12px"
    >
      This is the Toggle
    </label>
    <label
      class="c2"
      data-testid="This is the Toggle"
    >
      <span
        class="c3"
      />
    </label>
  </div>
</div>
`;

exports[`onClick Toggle 2`] = `
.c0 {
  display: flex;
  flex-direction: row;
  width: 100%;
  opacity: 1;
}

.c1 {
  font-family: Rubik-Bold;
  font-size: 12px;
  margin-right: 8px;
}

.c2 {
  position: relative;
  display: inline-block;
  height: 14px;
  background: #D41F3450;
  border-radius: 100px;
  width: 42.272px;
}

.c3 {
  flex-shrink: 0;
  fill: var(--system-colors-scl-white, #fff);
  filter: drop-shadow(0px 1.0458389520645142px 2.0916779041290283px rgba(0, 0, 0, 0.2)) drop-shadow(0px 0.10458389669656754px 0.3137516975402832px rgba(0, 0, 0, 0.1));
  border-radius: 50%;
  position: absolute;
  cursor: pointer;
  top: -6px;
  left: 19px;
  bottom: 0;
  background-color: #ccc;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  content: '';
  height: 26px;
  width: 26px;
  background-color: #D41F34;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
      font-family="Rubik-Bold"
      font-size="12px"
    >
      This is the Toggle
    </label>
    <label
      class="c2"
      data-testid="This is the Toggle"
    >
      <span
        class="c3"
      />
    </label>
  </div>
</div>
`;
